homeassistant/components/essent/sensor.py

```diff
@@ -0,0 +1,117 @@
+"""Support for Essent API."""
+from datetime import timedelta
+
+from pyessent import PyEssent
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (
+    CONF_PASSWORD, CONF_USERNAME, ENERGY_KILO_WATT_HOUR)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+SCAN_INTERVAL = timedelta(hours=1)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Essent platform."""
+    username = config[CONF_USERNAME]
+    password = config[CONF_PASSWORD]
+
+    essent = EssentBase(username, password)
+    add_devices(essent.retrieve_meters(), True)
+
+
+class EssentBase():
+    """Essent Base."""
+
+    def __init__(self, username, password):
+        """Initialize the Essent API."""
+        self._username = username
+        self._password = password
+        self._meters = []
+        self._meter_data = {}
+
+        self.update()
+
+    def get_session(self):
+        """Return the active session."""
+        return self._essent
+
+    def retrieve_meters(self):
+        """Retrieve the IDs of the meters used by Essent."""
+        meters = []
+        for meter in self._meters:
+            data = self._meter_data[meter]
```

Comment:
Is this right ?

Reply:
That call is, but the next call wasn't. Good catch, messed that up while restructuring it.